### DISCUSSION TITLE
Freetz for 7590 and FRITZ!OS 7.50

### DISCUSSION THREAD

**Discussion by Bars75** [Freetz-NG/freetz-ng#673] [Allgemein]
Hi!
When building freetz, the kernel option in the make/kernel/configs/freetz/config-grx5-7590_07.50 file is disabled:

# CONFIG_NFSD is not installed

Because of this, the image is not collected when nfsd is enabled.

If enabling the option to detect the latest firmware version:

Freetz-NG 20645M-48766008f master 2022-12-14
JUIS: Using cached value ... No cached value, asking avm for latest firmware ... no valid answer, trying backup ... http://download.avm.de/fritzbox/fritzbox-7590/other/fritz.os/FRITZ.Box_7590-07.29.image ... done.

STEP 1: UNPACK
source firmware file: dl/fw/FRITZ.Box_7590-07.29.image
checking signature: valid
unpacking firmware image
unpacking filesystem image

processing kernel
  checking kernel version
    ... used by AVM ...... 4.9.198
    ... used by Freetz ... 4.9.218
ERROR: Freetz and AVM kernel versions do not match. Wrong kernel version configuration in Freetz?
Are GPL sources for YOUR device and firware version released at http://osp.avm.de/ yet?
Have YOU asked fritzbox_info@avm.de for sources?

Thank you for your work!

**Comment by fda77** [comment]
AVM's update service "juis" is very crappy, it returns very often bad values. Check the history of daily checks: https://github.com/Freetz-NG/freetz-ng/commits/master/docs/juis/README.md
So there is nothing to do - excapt asking avm

**Comment by Bars75** [comment]
Well, it's okay, you can always turn off this option.

Please, you can add options to the kernel configuration make/kernel/configs/freetz/config-grx5-7590_07.50:

CONFIG_NFSD=m
CONFIG_NFSD_V3=y

Thanks!
